homeassistant/components/tuneblade_remote/coordinator.py

```diff
@@ -0,0 +1,49 @@
+"""Integration to integrate TuneBlade Remote devices with Home Assistant."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from aiohttp import ClientError
+from pytuneblade import TuneBladeApiClient
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TuneBladeDataUpdateCoordinator(DataUpdateCoordinator[dict[str, dict[str, Any]]]):
+    """Coordinator to fetch data from the TuneBlade hub."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: TuneBladeApiClient,
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=30),
+        )
+        self.client = client
+        self.data: dict[str, dict[str, Any]] = {}
```

Comment:
No need to initialize data as a dict, `_async_update_data` will be run at start so we will have a value anyway